processor/src/main/resources/org/mapstruct/ap/internal/model/IterableMappingMethod.ftl

```diff
@@ -54,7 +54,8 @@
 
     <#if resultType.arrayType>
         <#if !existingInstanceMapping>
-            <@includeModel object=resultElementType/>[] ${resultName} = new <@includeModel object=resultElementType/>[<@iterableSize/>];
+            <#assign elementTypeString><@includeModel object=resultElementType/></#assign>
```

Comment:
Yeah, that was my first thought as well. But I would either have to pull in the `iterableSize` macro into the java code as well, or provide explicit properties for the prefix and suffix... both seemed to blur the separation of responsibilities quite a lot... Or do you have a nice idea on how to split this?